read.js

```diff
@@ -120,12 +133,19 @@ class Entry {
     const start = this.offset + this.filenameLength + 30 + (extra ? extra + 4 : 0);
     const end = start + this.compressedSize;
 
-    return this
+    let stream = this
       ._fileLike
       .slice(start, end)
       .stream();
-      // .pipeThrought(inflate) // TODO: optional inflate
-      // .pipeThrought(crc) // TODO: crc32 validate
+
+    if (this.compressionMethod) {
```

Comment:
think i'm reading the wrong data, think there is some deflated bit flag somewhere in there somewhere

Reply:
Should I still merge for now?

Reply:
was right to read the compressionMethod... jszip and zip.js base it upon compressionMethod also

But we maybe should be more explicit about what kind of compression method it is and do things accordingly?

```
      compression method: (2 bytes)

          (see accompanying documentation for algorithm
          descriptions)

          0 - The file is stored (no compression)
          1 - The file is Shrunk
          2 - The file is Reduced with compression factor 1
          3 - The file is Reduced with compression factor 2
          4 - The file is Reduced with compression factor 3
          5 - The file is Reduced with compression factor 4
          6 - The file is Imploded
          7 - Reserved for Tokenizing compression algorithm
          8 - The file is Deflated
          9 - Enhanced Deflating using Deflate64(tm)
         10 - PKWARE Data Compression Library Imploding (old IBM TERSE)
         11 - Reserved by PKWARE
         12 - File is compressed using BZIP2 algorithm
         13 - Reserved by PKWARE
         14 - LZMA (EFS)
         15 - Reserved by PKWARE
         16 - Reserved by PKWARE
         17 - Reserved by PKWARE
         18 - File is compressed using IBM TERSE (new)
         19 - IBM LZ77 z Architecture (PFS)
         97 - WavPack compressed data
         98 - PPMd version I, Rev 1
```

but it's good for now i think